dao/src/main/java/se/fortnox/reactivewizard/db/ObservableStatementFactory.java

```diff
@@ -59,15 +59,15 @@ private static void closeSilently(Connection connection) {
 
     public Observable<Object> create(Object[] args, ConnectionProvider connectionProvider) {
         AtomicReference<TransactionStatement> transactionHolder = new AtomicReference<>();
+        Statement dbStatement = statementFactory.create(args);
```

Comment:
I guess this is important - but I don't really understand why you need to create the statement here. This code will be running on the calling thread from the users code. Moving the creation here might change how side-effects on the `args` param will affect the running query.

The main point of this PR is to avoid running the inner Observable here? Then I guess the whole block with if `(transactionStatement != null) {...}` would not be needed any more? Then transactionHolder is not needed as well?

Reply:
Exactly, the main point is to get rid of the inner observable. But I had to keep it in order to still support DaoTransactions.createTransaction.